packages/quantic/force-app/main/default/lwc/quanticInsightSummary/__tests__/quanticInsightSummary.test.js

```diff
@@ -0,0 +1,225 @@
+const initialQuerySummaryState = {hasError: false};
+let querySummaryState = initialQuerySummaryState;
+
+const exampleEngine = {
+  id: 'example engine',
+};
+
+const defaultOptions = {
+  engineId: exampleEngine.id,
+};
+
+const functionsMocks = {
+  buildSearchBox: jest.fn(() => ({
+    state: searchBoxState,
+    subscribe: functionsMocks.searchBoxStateSubscriber,
+  })),
+  buildQuerySummary: jest.fn(() => ({
+    state: querySummaryState,
+    subscribe: functionsMocks.querySummaryStateSubscriber,
+  })),
+  searchBoxStateSubscriber: jest.fn((cb) => {
+    cb();
+    return functionsMocks.searchBoxStateUnsubscriber;
+  }),
+  querySummaryStateSubscriber: jest.fn((cb) => {
+    cb();
+    return functionsMocks.querySummaryStateUnsubscriber;
+  }),
+  searchBoxStateUnsubscriber: jest.fn(),
+  querySummaryStateUnsubscriber: jest.fn(),
+};
+
+const selectors = {
+  initializationError: 'c-quantic-component-error',
+  initialSummaryMessage: '[data-testid="initial-summary-message"]',
+  quanticSummary: 'c-quantic-summary',
+};
+
+function prepareHeadlessState() {
+  // @ts-ignore
+  mockHeadlessLoader.getHeadlessBundle = () => {
+    return {
+      buildSearchBox: functionsMocks.buildSearchBox,
+      buildQuerySummary: functionsMocks.buildQuerySummary,
+    };
+  };
+}
+
+function mockSuccessfulHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element, _, initialize) => {
+    if (element instanceof QuanticInsightSummary && !isInitialized) {
+      isInitialized = true;
+      initialize(exampleEngine);
+    }
+  };
+}
+
+function mockErroneousHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element) => {
+    if (element instanceof QuanticInsightSummary) {
+      element.setInitializationError();
+    }
+  };
+}
+
+const createTestComponent = buildCreateTestComponent(
+  QuanticInsightSummary,
+  'c-quantic-insight-summary',
+  {defaultOptions}
+);
+
+describe('c-quantic-insight-summary', () => {
+  beforeEach(() => {
+    mockSuccessfulHeadlessInitialization();
+    prepareHeadlessState();
+  });
+
+  afterEach(() => {
+    cleanup();
+    searchBoxState = initialSearchBoxState;
+    querySummaryState = initialQuerySummaryState;
+    isInitialized = false;
+  });
+
+  describe('component initialization', () => {
+    it('should initialize the search box and query summary headless controllers', async () => {
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionsMocks.buildSearchBox).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.buildSearchBox).toHaveBeenCalledWith(exampleEngine);
+      expect(functionsMocks.buildQuerySummary).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.buildQuerySummary).toHaveBeenCalledWith(
+        exampleEngine
+      );
+    });
+
+    it('should subscribe the search box and query summary state changes', async () => {
+      createTestComponent();
+      await flushPromises();
+    });
```

Comment:
I think you're missing some stuff here :D

Reply:
thanks! will add!